app/view_utils/new_layout_view_utils.rb

```diff
@@ -38,18 +57,46 @@ def features(community_id, person_id)
   # and returns the keys as symbols from the entries
   # that hold value "true".
   def enabled_features(feature_params)
-    allowed_features = FEATURES.map { |f| f[:name] }
-    feature_params.select { |key, value| value == "true" }
-      .keys
-      .map(&:to_sym)
-      .select { |k| allowed_features.include?(k) }
+    allowed_features = (FEATURES + EXPERIMENTAL_FEATURES.values).map { |f| f[:name] }
+    features = feature_params.select { |key, value| value == "true" }
+                 .keys
+                 .map(&:to_sym)
+                 .select { |k| allowed_features.include?(k) }
+    add_required_features(features)
   end
 
   # From the list of features, selects the ones
   # that are disabled, ie. not included in the
   # list of enabled features.
   def resolve_disabled(enabled)
-    FEATURES.map { |f| f[:name]}
-      .select { |f| !enabled.include?(f) }
+     all_enabled = add_required_features(enabled)
+     features = (FEATURES + EXPERIMENTAL_FEATURES.values).map { |f| f[:name]}
+       .select { |f| !all_enabled.include?(f) }
+  end
+
+  # private
+
+  def add_required_features(features)
+    (features | FEATURE_RELS.values_at(*features)).compact
+  end
+
+
+  def all_features(person_flags, community_flags, private_community, clp_enabled)
+    all_flags = (person_flags | community_flags).to_a
+
+    # Additional rules for flags
+    all_flags.delete(:searchpage_v1) unless can_manage_searchpage?(all_flags, private_community, clp_enabled)
+
+    (FEATURES + EXPERIMENTAL_FEATURES.values_at(*all_flags)).compact
```

Comment:
Does this work? I assume that values_at should be applied to end-result of *FEATURES + EXPERIMENTAL_FEATURES* ? Now, I read this so that it's added to experimental_features only